addons/trenches/functions/fnc_continueDiggingTrench.sqf

```diff
@@ -63,29 +67,68 @@ private _fnc_onFailure = {
+private _fnc_condition = {
+   (_this select 0) params ["", "_trench"];
 
-if(_actualProgress == 0) then {
-    [_unit, _trench, _trenchId, _basePos vectorDiff [0, 0, 1.0], _vecDirAndUp, _actualProgress] call FUNC(setTrenchPlacement);
+   if !(_trench getVariable [QGVAR(digging), false]) exitWith {false};
+   if (_trench getVariable [QGVAR(diggerCount), 0] <= 0) exitWith {false};
+   if (GVAR(stopBuildingAtFatigueMax) && (EGVAR(advanced_fatigue,anReserve) <= 0))  exitWith {false};
+   true
+};
+
+[[_unit, _trench], _fnc_onFinish, _fnc_onFailure, localize LSTRING(DiggingTrench), _fnc_condition] call FUNC(progressBar);
 
-    //Remove grass
+if (_actualProgress == 0) then {
+      //Remove grass
     {
         private _trenchGrassCutter = createVehicle ["Land_ClutterCutter_medium_F", [0, 0, 0], [], 0, "NONE"];
         private _cutterPos = AGLToASL (_trench modelToWorld _x);
         _cutterPos set [2, getTerrainHeightASL _cutterPos];
         _trenchGrassCutter setPosASL _cutterPos;
         deleteVehicle _trenchGrassCutter;
-    } foreach getArray (configFile >> "CfgVehicles" >> (typeof _trench) >> QGVAR(grassCuttingPoints));
+    } foreach getArray (configFile >> "CfgVehicles" >> (typeOf _trench) >> "ace_trenches_grassCuttingPoints");
 };
 
-private _progressLeft = (_actualProgress * 10) + 1;
+[{
+  params ["_args", "_handle"];
+  _args params ["_trench", "_unit", "_digTime", "_vecDirAndUp"];
+  private _actualProgress = _trench getVariable [QGVAR(progress), 0];
+  private _diggerCount = _trench getVariable [QGVAR(diggerCount), 0];
+
+  if (
+        !(_trench getVariable [QGVAR(digging), false]) ||
+        (_diggerCount <= 0) ||
+        (_actualProgress >= 1)
+     ) exitWith {
+    [_handle] call CBA_fnc_removePerFrameHandler;
+    _trench setVariable [QGVAR(digging), false, true];
+    _trench setVariable [QGVAR(diggerCount), ((_diggerCount -1) max 0), true];
+  };
 
-for "_i" from _progressLeft to 10 do {
-    private _vectorDiffZ = 1 - (_i / 10);
-    private _delay = _digTime * ((_i / 10) - _actualProgress);
-    private _progress = _i / 10;
-    [DFUNC(setTrenchPlacement), [_unit, _trench, _trenchId, _basePos vectorDiff [0, 0, _vectorDiffZ], _vecDirAndUp, _progress], _delay] call CBA_fnc_waitAndExecute;
-};
+  private _boundingBox = boundingBoxReal _trench;
+  _boundingBox params ["_lbfc"];                                         //_lbfc(Left Bottom Front Corner) _rtbc (Right Top Back Corner)
+  _lbfc params ["", "", "_lbfcZ"];
+
+  private _pos = (getPosWorld _trench);
+  private _posDiff = ((abs((_trench getVariable [QGVAR(diggingSteps), 0]) + _lbfcZ)) * _diggerCount)/(_digTime*5);
+  _pos set [2,((_pos select 2) + _posDiff)];
+
+  _trench setPosWorld _pos;
+  _trench setVectorDirAndUp _vecDirAndUp;
+
+  //Fatigue impact
+  EGVAR(advanced_fatigue,anReserve) = (EGVAR(advanced_fatigue,anReserve) - ((_digTime /12) * GVAR(buildFatigueFactor))) max 0;
```

Comment:
useless parenthesis. macro is resolved before anything happens here.
More useless parens in L121, L124, L126, L129